packages/gatsby-plugin-manifest/src/gatsby-node.js

```diff
@@ -69,6 +73,18 @@ exports.onPostBootstrap = async (args, pluginOptions) => {
       throw `icon (${icon}) does not exist as defined in gatsby-config.js. Make sure the file exists relative to the root of the site.`
     }
 
+    let sharpIcon = sharp(icon)
+
+    sharpIcon.metadata().then(metadata => {
+      if (metadata.width !== metadata.height) {
+        reporter.warn(
+          `The icon(${icon}) you provided to 'gatsby-plugin-manifest' is not square. \n` +
+            `The icons we generate will be square and for the best results we recommend you provide a square icon. \n` +
+            `We'll do our best with what you provide :)`
```

Comment:
I would drop this sentence but maybe @DSchau thinks differently.
```suggestion
            `We'll do our best with what you provide :)`
```

Reply:
Agreed with ya! Let's drop this!

Reply:
So I'm open to reword this but I feel something is needed. The previous sentences make it sound like something is broken or wrong. But that's not the case, the generation has suceeded. We're just giving them a heads-up. Thoughts?

Reply:
Nevermind...I just re-read it and it's fine without.